lib/boot/boot.go

```
@@ -0,0 +1,149 @@
+	"sync"
+	"syscall"
+)
+
+type App struct {
+	ctx             context.Context
+	cfg             *config.Config
+	secret          *secrets.Secrets
+	grpcServer      *grpc.Server
+	grpcRegister    func(*grpc.Server)
+	db              *postgres.Connection
+	tx              *transaction_manager.TransactionManager
+	syncProducer    sarama.SyncProducer
+	consumerManager *consumer.ConsumerManager
+}
+
+type Option func(*App) error
+
+func NewApp(ctx context.Context, opts ...Option) (*App, error) {
+	app := &App{
+		ctx: ctx,
+	}
+	for _, opt := range opts {
+		if err := opt(app); err != nil {
+			return nil, fmt.Errorf("failed to apply option: %w", err)
+		}
+	}
+
+	if app.cfg == nil {
+		return nil, errors.New("config is required")
+	}
+
+	app.grpcServer = grpc.NewServer(
+		grpc.Creds(insecure.NewCredentials()),
+		grpc.ChainUnaryInterceptor(interceptors.ProtoValidate),
+	)
+
+	return app, nil
+}
+
+// WithConfigXXX конфиг с интегрированными секретами, ENV переменными, значениями по умолчанию
+func WithConfigXXX(ctx context.Context, cfg Config) Option {
+	return func(app *App) error {
+		cl := config.NewConfigLoader().Configure(
+			config.WithDefaults(cfg.DefaultValue),
+			config.WithEnvBindings(cfg.EnvBinding),
+		)
+
+		if cfg.SecretProvider != nil {
+			secret := secrets.NewSecrets(cfg.SecretProvider)
+			if secret.IsSet() {
+				cl.WithSecretsProvider(secret)
+				cl.AddSecretKeys(cfg.SecretKeys...)
+			}
+		}
+
+		cfg, err := cl.LoadConfig(cfg.ConfigFile)
+		if err != nil {
+			return fmt.Errorf("failed to load config: %w", err)
+		}
+
+		app.cfg = cfg
+		return nil
+	}
+}
+
+func (app *App) Run(ctx context.Context) error {
+	if app.grpcRegister != nil {
+		app.grpcRegister(app.grpcServer)
+	}
+
+	if app.consumerManager != nil {
+		err := app.consumerManager.StartAll(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	lis, err := net.Listen("tcp", ":"+strconv.Itoa(app.cfg.Grpc.Port))
+	if err != nil {
+		return fmt.Errorf("failed to listen: %w", err)
+	}
```

Вообще листенер уже можно. и нужно на создании еще NewApp (принцип fail fast - если порт занят - упадем сразу)

Также позволит liveness/startup пробе сразу отработать